pkg/template/event.go

```diff
@@ -96,6 +97,18 @@ func applyEventValuesToParams(params []triggersv1.Param, body []byte, header htt
 		expressions, originals := findTektonExpressions(pValue)
 		for i, expr := range expressions {
 			val, err := ParseJSONPath(event, expr)
+			if defaults != nil && err != nil {
+				// if the header or body was not supplied or was malformed, go with a default if it exists
+				for _, ps := range defaults {
+					if ps.Name == p.Name && ps.Default != nil {
+						fmt.Fprintf(os.Stdout, "using default %s since failed to replace JSONPath value for param %s: %s: error: %s\n",
```

Comment:
This is useful information, but I'm not sure we need to print it out. If we do, maybe we can pass through the logger that we for other logging instead of using fmt.Fprintf?

Reply:
Yeah I looked at that and found it a bit onerous to pass the logger through all the call levels to get it here.

But if this is not an tenable compromise, then sure.  I can go one way or the other.  I don't have a strong opinion.  Will it be feasible for a user to figure out that if the default is used vs. the body/header reference, there was an issue with how it was constructed?

You got a final call here @dibyom ?

Reply:
Yeah, I think its fine to remove the log statement here given that this is documented/expected behavior.

Reply:
ok just pushed removal of log statement as part of rebasing this PR on top of recent changes